### ISSUE TITLE
proof of humanity

### ISSUE THREAD

**Issue by posix4e** [polkadot-js/api#4160]
We are curious about adding proof of humanity to polkajs so we can provide onchain proof that a human performed the transaction. Just like to https://www.humanprotocol.org/blog/proof-of-humanity-human-and-metamask-integration?lng=en-US

**Comment by jacogr** [comment]
The API will auto-decorate on any chain it connects to to expose the interfaces as defined by the metadata. Basically it has no hard-coded interfaces, all is driven by the on-chain runtime metadata.

**Comment by posix4e** [comment]
Huh, that's interesting. I know human protocol is interested in adding it to polkajs so that all the wallets inherit it. We could do it external to the wallet but I see it as cool feature to require a transaction to be done human - human

**Comment by posix4e** [comment]
Do we have some examples of updating these interfaces?

**Comment by jacogr** [comment]
So, let me point to the docs for the metadata - https://polkadot.js.org/docs/api/start/basics#metadata

So how the API works -

1. the skeleton is completely empty
2. when it connects to a chain it retrieves the metadata from the runtime
3. based on this metadata the interfaces that are available are populated
4. these interfaces (as per the on-chain metadata, different for each Substrate-based chain) is then available to the user

So basically anything that is exposed by the on-chain runtime is available in the API - everything is fully dynamic, e.g. some chains have for instance `balances.transfer` to send funds, some chains may not have this at all. Some chains have democracy, some chains do not. It is basically a block on LEGOs in the substrate world and the API detects which pieces are there by getting the on-chain metadata.
